deps/getdns/getdns.gyp

```diff
@@ -0,0 +1,145 @@
+{
+  'targets': [
+    {
+      'target_name': 'getdns',
+      'type': 'static_library',
+      'include_dirs': [
+        'getdns/src',
+        'getdns/src/getdns',
+        'getdns/src/util/auxiliary',
+        'getdns/src/openssl',
+        'getdns/src/tls',
+        'getdns/src/yxml',
+      ],
+      'direct_dependent_settings': {
+        'include_dirs': [
+          'getdns/src',
+        ],
+      },
+      'conditions': [
+        ['node_shared_openssl=="false"', {
+          'dependencies': [
+            '../openssl/openssl.gyp:openssl'
+          ]
+        }],
+        ['node_shared_libuv=="false"', {
+          'dependencies': [
+            '../uv/uv.gyp:libuv'
+          ]
+        }],
+        [ 'OS=="linux"', {
+          'defines': [
+            '_GNU_SOURCE',
+            '_DEFAULT_SOURCE',
+          ],
+          'sources': [
+            'getdns/src/compat/arc4random.c',
+            'getdns/src/compat/explicit_bzero.c',
+            'getdns/src/compat/arc4_lock.c',
+            'getdns/src/compat/arc4random_uniform.c',
+            'getdns/src/compat/strlcpy.c',
+          ],
+        }],
+        [ 'OS=="win"', {
+          'sources': [
+            'getdns/src/compat/gettimeofday.c',
+            'getdns/src/compat/arc4random.c',
+            'getdns/src/compat/arc4_lock.c',
+            'getdns/src/compat/arc4random_uniform.c',
+            'getdns/src/compat/getentropy_win.c',
+            'getdns/src/compat/strlcpy.c',
+            'getdns/src/compat/mkstemp.c',
+            'getdns/src/compat/strptime.c',
+          ],
+        }],
+        [ 'OS=="freebsd"', {
+          'defines': [
+            '_POSIX_C_SOURCE=200112L',
+            '_XOPEN_SOURCE=600',
+          ],
+        }],
+        [ 'OS=="solaris"', {
+          'defines': [
+            '__EXTENSIONS_',
+          ],
+        }],
+        [ 'OS in "mac ios"', {
+          'defines': ['_DARWIN_C_SOURCE'],
+        }],
+      ],
+      'actions': [
+        {
+          'action_name': 'configure_file',
+          'process_outputs_as_sources': 1,
+          'inputs': [
+            # Put the code first so it's a dependency and can be used for invocation.
+            'configure_file.py',
+            'getdns/cmake/include/cmakeconfig.h.in',
+            'getdns/src/getdns/getdns.h.in',
+            'getdns/src/getdns/getdns_extra.h.in',
+            'getdns/src/version.c.in',
+            'getdns/getdns.pc.in',
+          ],
+          'outputs': [
+            'getdns/src/config.h',
+            'getdns/src/getdns/getdns.h',
+            'getdns/src/getdns/getdns_extra.h',
+            'getdns/src/version.c',
+            'getdns/getdns.pc',
```

Comment:
Usually the outputs would go into `<(INTERMEDIATE_DIR)` or `<(SHARED_INTERMEDIATE_DIR)`.

Reply:
this is mirroring the output of the cmake config. is it related to the cause of the ci failures?

Reply:
Yes.

```diff
diff --git a/deps/getdns/getdns.gyp b/deps/getdns/getdns.gyp
index f035598dd0..fb13a83bda 100644
--- a/deps/getdns/getdns.gyp
+++ b/deps/getdns/getdns.gyp
@@ -10,10 +10,12 @@
         'getdns/src/openssl',
         'getdns/src/tls',
         'getdns/src/yxml',
+        '<(SHARED_INTERMEDIATE_DIR)/getdns/src',
       ],
       'direct_dependent_settings': {
         'include_dirs': [
           'getdns/src',
+          '<(SHARED_INTERMEDIATE_DIR)/getdns/src',
         ],
       },
       'conditions': [
@@ -81,11 +83,11 @@
             'getdns/getdns.pc.in',
           ],
           'outputs': [
-            'getdns/src/config.h',
-            'getdns/src/getdns/getdns.h',
-            'getdns/src/getdns/getdns_extra.h',
-            'getdns/src/version.c',
-            'getdns/getdns.pc',
+            '<(SHARED_INTERMEDIATE_DIR)/getdns/src/config.h',
+            '<(SHARED_INTERMEDIATE_DIR)/getdns/src/getdns/getdns.h',
+            '<(SHARED_INTERMEDIATE_DIR)/getdns/src/getdns/getdns_extra.h',
+            '<(SHARED_INTERMEDIATE_DIR)/getdns/src/version.c',
+            '<(SHARED_INTERMEDIATE_DIR)/getdns/getdns.pc',
           ],
           'action': [
             'python', '<@(_inputs)', '--target', '<@(_outputs)',
@@ -139,6 +141,7 @@
         'getdns/src/openssl/tls.c',
         'getdns/src/openssl/pubkey-pinning-internal.c',
         'getdns/src/openssl/keyraw-internal.c',
+        '<(SHARED_INTERMEDIATE_DIR)/getdns/src/version.c',
       ]
     }
   ]
```
allows me to build from a clean repository (`git clean -fdX`). I get 10 test failures but I assume those are not unexpected/being worked out.

Reply:
thank you!

Reply:
I haven't tested it but it's possible that the non-headers (i.e. `getdns/src/version.c` and `getdns/getdns.pc`) could go in `<(INTERMEDIATE_DIR)` as they probably aren't needed be available to the things linking to `getdns` (whereas the generated headers are and need to go into the shared area).